lib/puma/cluster/worker_handle.rb

```diff
@@ -51,7 +51,9 @@ def term?
         @term
       end
 
-      STATUS_PATTERN = /{ "backlog":(?<backlog>\d*), "running":(?<running>\d*), "pool_capacity":(?<pool_capacity>\d*), "max_threads":(?<max_threads>\d*), "requests_count":(?<requests_count>\d*) }/
+      regex_string = Puma::Server::STAT_METHODS.map { |s| "\"#{s}\":(?<#{s}>\\d*)"  }.join(', ')
```

Comment:
I like this code, it's much cleaner and it's not on the hot path. 

However, I think this will also have to be reverted if you change the above method of stats collection unless we want to normalize the order of the values (such as alphabetical). I'm assuming we have tests that exercise this status pattern.

Reply:
sounds good

fwiw the particular order is already necessary because worker.rb and worker_handler.rb have a producer/consumer relationship

Reply:
reverted